crates/sound/src/audio_backends/pw_backend.rs

```diff
@@ -95,7 +85,10 @@ impl AudioBackend for PwBackend {
         */
         Ok(())
     }
-    fn set_param(&self, _stream_id: u32, _params: PCMParams) -> Result<()> {
+    fn set_param(&self, stream_id: u32, params: PCMParams) -> Result<()> {
+        let mut stream_params = self.stream_params.write().unwrap();
+        stream_params[stream_id as usize] = params;
```

Comment:
This line is triggering: 
```
thread 'vring_worker' panicked at 'index out of bounds: the len is 0 but the index is 0'
```

Reply:
IIUC, the stream_params.len() is 0 here so accessing stream_params[0] would not work.

Reply:
I wrongly though that when instantiating a vector `withcapacity()`, it would allow me to access the vector by using an index. It seems that I have to do `vec.resize(len, 0)` after the instantiation to then be able to access by index.

Reply:
@MatiasVara you can use the `vec!` macro: https://doc.rust-lang.org/std/macro.vec.html